etna/analysis/feature_selection/mrmr.py

```diff
@@ -0,0 +1,59 @@
+from typing import List
+
+import numpy as np
+import pandas as pd
+from sklearn.feature_selection import f_classif
+
+FLOOR = 0.00001
+
+
+def mrmr(x: pd.DataFrame, y: np.ndarray, top_k: int) -> List[str]:
+    """
+    Maximum Relevance and Minimum Redundancy feature selection method.
+
+    Parameters:
+    ----------
+    x:
+        dataframe of shape n_segment x n_exog_series with relevance table, where relevance_table[i][j] contains relevance
+        of j-th df_exog series to i-th df series
+    y:
+        class(cluster) labels of the segments
+    top_k:
+        num of regressors to select; if there are not enough regressors, then all will be selected
+
+    Returns:
+    -------
+    selected_features: List[str]
+        list of `top_k` selected regressors, sorted by their importance
+    """
+    x = x.dropna(axis=1, how="all")
+    relevance_table = x.apply(lambda col: f_classif(col[~col.isna()].to_frame(), y[~col.isna()])[0][0])
+    relevance_table = relevance_table[relevance_table > 0]
```

Comment:
why do you need this filter?

Reply:
It count relevance for all regressors separately and keep only regressors that have positive relevance

Reply:
is it just about optimization of number of series we work with?

Reply:
This is how it was implemented in mrmr library)